src/lib/functions/server.js

```diff
@@ -89,7 +89,8 @@ const createHandler = function (options) {
       {},
     )
     const rawQuery = new URLSearchParams(request.query).toString()
-    const url = new URL(requestPath, `${request.protocol}://${request.get('host') || 'localhost'}`)
+    const protocol = options.config.dev.https ? 'https' : 'http'
+    const url = new URL(requestPath, `${protocol}://${request.get('host') || 'localhost'}`)
```

Comment:
so the server doesn't listen on 80 anymore when you enable https?

Reply:
We have multiple servers running: a main server that then forwards requests to a functions server. The main server is listening on 443, the functions server is listening on 80.

This will continue to be the case, all we're changing is the event data that is sent to functions. Rather than using the protocol from the functions server (which will always be 80), we're using the protocol of the main server.

Reply:
but if you have two servers, you can't really use the config setting right?
since you need to set the protocol based on which server the request came in.
i'd recommend using a HTTP header for this.

Reply:
Only the main/outer server is exposed to customers. The functions server is an internal implementation detail. So the `rawUrl` value of functions should always match the URL of the outer server.